Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -341,32 +453,263 @@ define([
             result);
     };
 
-    Cesium3DTileBatchTable.prototype.hasProperty = function(name) {
+    function getBinaryProperty(binaryProperty, index) {
+        var typedArray = binaryProperty.typedArray;
+        var componentCount = binaryProperty.componentCount;
+        if (componentCount === 1) {
+            return typedArray[index];
+        } else {
+            return binaryProperty.type.unpack(typedArray, index * componentCount);
+        }
+    }
+
+    function setBinaryProperty(binaryProperty, index, value) {
+        var typedArray = binaryProperty.typedArray;
+        var componentCount = binaryProperty.componentCount;
+        if (componentCount === 1) {
+            typedArray[index] = value;
+        } else {
+            binaryProperty.type.pack(value, typedArray, index * componentCount);
+        }
+    }
+
+    var scratchStack = [];
+    var marker = 0;
```

Comment:
If the marker is reset then the array also needs to be reset, which I'm trying to avoid. Rather than resetting I just increment it so on the second call tiles that are visited are marked 2, on the third call they are marked as 3, etc.